src/org/opensolaris/opengrok/configuration/messages/ConfigMessage.java

```diff
@@ -33,16 +33,36 @@
 
     @Override
     protected byte[] applyMessage(RuntimeEnvironment env) throws IOException {
-        env.applyConfig(this, hasTag("reindex"));
+        if (hasTag("getconf")) {
+            return env.getConfiguration().getXMLRepresentationAsString().getBytes();
+        } else if (hasTag("setconf")) {
+            env.applyConfig(this, hasTag("reindex"));
+        }
 
         return null;
     }
 
     @Override
     public void validate() throws Exception {
-        if (getText() == null) {
-            throw new Exception("The message must contain a text.");
+        if (hasTag("setconf")) {
+            if (getText() == null) {
+                throw new Exception("The setconf message must contain a text.");
+            }
+        } else if (hasTag("getconf")) {
```

Comment:
Somehow I am not thrilled by the idea of the app second guessing users' intentions.